diff --git a/node_modules/@gouvfr/dsfr/dist/dsfr/dsfr.module.js b/node_modules/@gouvfr/dsfr/dist/dsfr/dsfr.module.js
index de30a88..62e8889 100644
--- a/node_modules/@gouvfr/dsfr/dist/dsfr/dsfr.module.js
+++ b/node_modules/@gouvfr/dsfr/dist/dsfr/dsfr.module.js
@@ -3726,7 +3726,7 @@ class HeaderModal extends api.core.Instance {
       id = button.id || id;
       if (button.isPrimary && id) break;
     }
-    this.setAttribute('aria-labelledby', id);
+    if (id) this.setAttribute('aria-labelledby', id);
   }
 
   unqualify () {
